guides/fcm-flutter-novu/how-to-send-push-notifications-to-flutter-apps-with-fcm-using-Novu.mdx

```diff
@@ -0,0 +1,735 @@
+  const MyApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: 'Flutter Demo',
+      theme: ThemeData(
+        primarySwatch: Colors.blue,
+      ),
+      home: const MyHomePage(title: 'Flutter Demo Home Page'),
+    );
+  }
+}
+
+class MyHomePage extends StatefulWidget {
+  const MyHomePage({Key? key, required this.title}) : super(key: key);
+
+  final String title;
+
+  @override
+  State<MyHomePage> createState() => _MyHomePageState();
+}
+
+class _MyHomePageState extends State<MyHomePage> {
+  String _lastMessage = "";
+
+  _MyHomePageState() {
+    _messageStreamController.listen((message) {
+      setState(() {
+        if (message.notification != null) {
+          _lastMessage = 'Received a notification message:'
+              '\nTitle=${message.notification?.title},'
+              '\nBody=${message.notification?.body},'
+              '\nData=${message.data}';
+        } else {
+          _lastMessage = 'Received a data message: ${message.data}';
+        }
+      });
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(widget.title),
+      ),
+      body: Center(
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[
+            Text('Last message from Firebase Messaging:',
+                style: Theme.of(context).textTheme.titleLarge),
+            Text(_lastMessage, style: Theme.of(context).textTheme.bodyLarge),
+          ],
+        ),
+      ),
+    );
+  }
+}
+```
+_lib/main.dart_
+
+There are 4 sections to pay attention to:
+
+<Steps>
+  <Step title="The Request permission code block">
+    This shows the notification permission prompt to the user the first time the user interacts with the app. If the user allows it, then we can get a token.
+
+    ```dart
+    // Request permission
+    final messaging = FirebaseMessaging.instance;
+
+    // Web/iOS app users need to grant permission to receive messages
+    final settings = await messaging.requestPermission(
+      alert: true,
+      announcement: false,
+      badge: true,
+      carPlay: false,
+      criticalAlert: false,
+      provisional: false,
+      sound: true,
+    );
+    ```
+
+  </Step>
+  <Step title="Registering and obtaining the FCM token for the device">
+    A device's token is needed for the specific device to receive messages. The `getToken()` returns the registration token for the app instance on the device.
+
+    ```dart
+      String? token = await messaging.getToken();
+
+      if (kDebugMode) {
+        print('Registration Token=$token');
+      }
+    ```
+  </Step>
+  <Step title="The foreground message handler">
+    This handler listens to when a push notification is fired from FCM to the device. If the app is in foreground, then it prints out the notification title, body, messageId and data properties to the console.
```

Comment:
1. This handler listens to when a push notification is fired from FCM to the device => This handler is triggered when a push notification is sent from FCM to the device.
2. If app is in foreground => if app is in the foreground
3. messageId and data => messageId, and data